src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs

```diff
@@ -177,23 +177,30 @@ public void Dispose()
         {
             foreach (CoreNode node in this.Nodes)
                 node.Kill();
-
-            KillAnyBitcoinInstances();
         }
 
         internal static void KillAnyBitcoinInstances()
         {
-            while (true)
+            var cancellationToken = new CancellationTokenSource(TimeSpan.FromSeconds(15)).Token;
+
+            while (!cancellationToken.IsCancellationRequested)
             {
-                Process[] bitcoinDProcesses = Process.GetProcessesByName("bitcoind");
-                IEnumerable<Process> applicableBitcoinDProcesses = bitcoinDProcesses.Where(b => b.MainModule.FileName.Contains("External Libs"));
-                if (!applicableBitcoinDProcesses.Any())
-                    break;
+                try
+                {
+                    Process[] bitcoinDProcesses = Process.GetProcessesByName("bitcoind");
+                    IEnumerable<Process> applicableBitcoinDProcesses =
+                        bitcoinDProcesses.Where(b => b.MainModule.FileName.Contains("External Libs"));
+
+                    if (!applicableBitcoinDProcesses.Any()) break;
 
-                foreach (Process process in applicableBitcoinDProcesses)
+                    foreach (Process process in applicableBitcoinDProcesses)
+                    {
+                        process.Kill();
+                        Thread.Sleep(100);
+                    }
+                }
+                catch
```

Comment:
I would say yes as killing the bitcoin core node is not sufficient to fail the test. Do let me know if you think otherwise and I will be happy to make necessary changes